### PR TITLE
bug-fix Character can jump on inclined surfaces

### PR DESCRIPTION
**Pull request by gabriellcarpes**
Using Character Controller to check Slope limit
If the character is currently above Slope Limit he will begin sliding down
When below the slope limit, he will stop sliding.
Sliding does not turn the character (but it could be if animation wants him to slide down his butt)

Related to issue: https://github.com/UnityTechnologies/open-project-1/issues/8 #
#8 

Card: https://open.codecks.io/unity-open-project-1/decks/13/card/16f-player-can-climb-steep-surfaces

### CODE REVIEW

File: UOP1_Project/Assets/Scripts/Characters/Character.cs [hunk 1]
```diff
@@ -104,6 +112,18 @@ private void OnControllerColliderHit(ControllerColliderHit hit)
     {
         bool isMovingUpwards = verticalMovement > 0f;
 
+        // Calculates if hit vector is above slopelimit set on Character Controller
+        isAboveSlopeLimit = (Vector3.Angle (Vector3.up, hit.normal) >= characterController.slopeLimit);
+        slideMovement = new Vector3(hit.normal.x, hit.normal.y, hit.normal.z);
+
+        if (isAboveSlopeLimit)
```
Comment (m4rtint) [inline UOP1_Project/Assets/Scripts/Characters/Character.cs:119]: can shorten it to

isSliding = isAboveSlopeLimit

Comment (gabriellcarpes) [reply UOP1_Project/Assets/Scripts/Characters/Character.cs]: You're right. I just changed it. Thanks :)

File: UOP1_Project/Assets/Scripts/Characters/Character.cs [hunk 2]
```diff
@@ -23,6 +23,9 @@ public class Character : MonoBehaviour
     private float verticalMovement = 0f; //Represents how much a player will move vertically in a frame. Affected by gravity * gravityContributionMultiplier
     private Vector3 inputVector; //Initial input horizontal movement (y == 0f)
     private Vector3 movementVector; //Final movement vector
+    private bool isAboveSlopeLimit = false; // Used to check if current ground is above Slope Limit set on Character Controller
+    public bool isSliding = false; // Detects if character is currently sliding down a slope
```
Comment (m4rtint) [inline UOP1_Project/Assets/Scripts/Characters/Character.cs:27]: there's no point of this being public, should just be private

Comment (gabriellcarpes) [reply UOP1_Project/Assets/Scripts/Characters/Character.cs]: Yeah, I changed it to public in order to see it better on the editor while developing and forgot to change it back

Comment (m4rtint) [reply UOP1_Project/Assets/Scripts/Characters/Character.cs]: you an also do 
[SerializedField]
private float blabla

and it'll show up on inspector, just a fyi .

**Comment by ciro-unity** [comment]
Sorry @gabriellcarpes, looks like we already fixed the bug so I'm inclined to close this, unless you think your fix is much better? :D :D

**Comment by gabriellcarpes** [comment]
> Sorry @gabriellcarpes, looks like we already fixed the bug so I'm inclined to close this, unless you think your fix is much better? :D :D

Hey Ciro. That's perfectly ok! I checked the PR that was already merged and I'm just glad that I was going on a similar direction to what was eventually merged

**Comment by ciro-unity** [comment]
Great. Don't worry, there will be more chances to get code in :)
Thanks for the contribution anyway, and to the reviewer @m4rtint !

**Pull request closed without merging**